core/domain/exp_domain.py

```diff
@@ -2461,6 +2560,33 @@ def get_all_html_content_strings(self):
 
         return html_list
 
+    def to_proto(self):
+        """Returns a proto representation of the exploration object.
+
+        Returns:
+            ExplorationDto. The proto object.
+        """
+        state_protos = {}
+        for (state_name, state) in self.states.items():
+            state_proto = state.to_proto()
+            state_protos[state_name] = state_proto
+
+        return exploration_pb2.ExplorationDto(
+            id=self.id,
+            content_version=self.version,
+            init_state_name=self.init_state_name,
+            title=self.title,
+            states=state_protos
```

Comment:
Maybe we can come up with some suffixes for field names as well. That will be very helpful same as what we do for message name. `Dto`

Reply:
OK, maybe leave this as a comment on PR 1 on oppia-proto-api if it makes sense, as well as have offline discussions with him, so that it doesn't get dropped. (Ditto for anything else here that says "ask Ben".)

Reply:
state_protos is a bit inconsistent compared to other proto fields, so I'm not super keen on it. Within the context of the proto, 'state_protos' will actually look odd.